### PR TITLE
Only import new codes ACDM-1581 #resolve

### PR DESCRIPTION
**Pull request by sergiofbsilva**

### CODE REVIEW

File: src/main/java/org/fenixedu/academic/ui/struts/action/manager/payments/SIBSPaymentsDA.java [hunk 1]
```diff
@@ -284,19 +285,21 @@ private void processCode(SibsIncommingPaymentFileDetailLine detailLine, Person p
...
-
-        if (codeToProcess.getState() == PaymentCodeState.INVALID) {
-            result.addMessage("warning.manager.SIBS.invalidCode", codeToProcess.getCode());
+        if (!(paymentCode instanceof EventPaymentCode)) {
+            result.addMessage("warning.manager.SIBS.outdated.code", paymentCode.getCode(), paymentCode.getExternalId());
+            return;
+        }
+        
+        if (paymentCode.getState() == PaymentCodeState.INVALID) {
+            result.addMessage("warning.manager.SIBS.invalidCode", paymentCode.getCode());
```
Comment (danielcorreia96) [inline src/main/java/org/fenixedu/academic/ui/struts/action/manager/payments/SIBSPaymentsDA.java]: @sergiofbsilva I think there's a _return_ missing here.

File: src/main/resources/resources/ManagerResources_en.properties [hunk 2]
```diff
@@ -1748,4 +1748,5 @@ info.manage.associated.objects.no.administrativeOffices=There are no administrat
 info.manage.associated.objects.no.emptyDegree=There is no empty degree.
 title.manage.associated.objects.departments.edit=Edit Department
 title.manage.associated.objects.administrativeOffices.create=Create Administrative Office
-info.manage.associated.objects.emptyDegree.associated.with=Empty degree associated with:
\ No newline at end of file
+info.manage.associated.objects.emptyDegree.associated.with=Empty degree associated with:
+warning.manager.SIBS.outdate.code=The payment code {0} ({1}) is old and should already be registered. Check this manually.
```
Comment (danielcorreia96) [inline src/main/resources/resources/ManagerResources_en.properties]: @sergiofbsilva There is a typo in the key here. It should be "warning.manager.SIBS.outdated.code" 🔍

File: src/main/resources/resources/ManagerResources_pt.properties [hunk 3]
```diff
@@ -1746,4 +1746,5 @@ info.manage.associated.objects.no.administrativeOffices=Não existem secretarias
 info.manage.associated.objects.no.emptyDegree=Não existe curso de unidades isoladas.
 title.manage.associated.objects.departments.edit=Editar Departamento
 title.manage.associated.objects.administrativeOffices.create=Criar Secretaria
-info.manage.associated.objects.emptyDegree.associated.with=Curso de unidades isoladas associado com:
\ No newline at end of file
+info.manage.associated.objects.emptyDegree.associated.with=Curso de unidades isoladas associado com:
+warning.manager.SIBS.outdate.code=O código de pagamento {0} ({1}) é antigo e já devia estar integrado. Analisar manualmente.
```
Comment (danielcorreia96) [inline src/main/resources/resources/ManagerResources_pt.properties]: @sergiofbsilva There is a typo in the key here. It should be "warning.manager.SIBS.outdated.code" 🔍

Comment (sergiofbsilva) [reply src/main/resources/resources/ManagerResources_pt.properties]: @danielcorreia96 e vires cá este ano não ? 😆

File: src/main/java/org/fenixedu/academic/domain/accounting/PaymentCode.java [hunk 4]
```diff
@@ -215,17 +215,12 @@ public final void process(Person responsiblePerson, SibsIncommingPaymentFileDeta
...
 
 
-    public boolean matches (SibsIncommingPaymentFileDetailLine sibsDetailLine) {
-        final Person person = getPerson();
-        if (person == null) {
-            return false;
-        }
-
-        return person.getEventsSet()
+    public boolean matches(SibsIncommingPaymentFileDetailLine sibsDetailLine) {
```
Comment (danielcorreia96) [inline src/main/java/org/fenixedu/academic/domain/accounting/PaymentCode.java]: @sergiofbsilva This method no longer uses anything from the PaymentCode class. It should probably be moved somewhere else (eventually).

**Pull request merged**